docs/audit-log-filter-advanced.md

```diff
@@ -0,0 +1,847 @@
+                }
+              }
+            },
+            "replace": {
+              "function": {
+                "name": "query_digest"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Multiple Field Replacement
+
+Replace values in different event classes:
+
+```json
+{
+  "filter": {
+    "class": [
+      {
+        "name": "general",
+        "print": {
+          "field": {
+            "name": "general_query.str",
+            "print": false,
+            "replace": {
+              "function": {
+                "name": "query_digest"
+              }
+            }
+          }
+        }
+      },
+      {
+        "name": "table_access",
+        "print": {
+          "field": {
+            "name": "query.str",
+            "print": false,
+            "replace": {
+              "function": {
+                "name": "query_digest"
+              }
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+## Dynamic Filter Replacement
+
+### Basic Filter Replacement
+
+Create filters that can change dynamically:
+
+```json
+{
+  "filter": {
+    "id": "main",
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "update",
+          "log": false,
+          "filter": {
+            "class": {
+              "name": "general",
+              "event": [
+                {
+                  "name": "status",
+                  "filter": {
+                    "ref": "main"
+                  }
+                }
+              ]
+            },
+            "activate": {
+              "or": [
+                {
+                  "field": {
+                    "name": "table_name.str",
+                    "value": "temp_1"
+                  }
+                },
+                {
+                  "field": {
+                    "name": "table_name.str",
+                    "value": "temp_2"
+                  }
+                }
+              ]
+            }
+          }
+        },
+        {
+          "name": "delete",
+          "log": false,
+          "filter": {
+            "class": {
+              "name": "general",
+              "event": [
+                {
+                  "name": "status",
+                  "filter": {
+                    "ref": "main"
+                  }
+                }
+              ]
+            },
+            "activate": {
+              "or": [
+                {
+                  "field": {
+                    "name": "table_name.str",
+                    "value": "temp_1"
+                  }
+                },
+                {
+                  "field": {
+                    "name": "table_name.str",
+                    "value": "temp_2"
+                  }
+                }
+              ]
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Conditional Filter Activation
+
+Activate different filters based on conditions:
+
+```json
```

Comment:
Not what was the idea here.
We define `"id": "sensitive_operations"` but never reference this `sensitive_operations` in any `"ref"`